### PR TITLE
samples: audio: add filter

### PR DESCRIPTION
**Pull request by nashif**
Limit this to only platforms that can use SOF.

### CODE REVIEW

**Comment by galak** [comment]
When I try this locally I get:

```
In file included from /home/galak/git/modules/audio/sof/src/platform/intel/cavs/platform.c:46:
/home/galak/git/modules/audio/sof/zephyr/../src/include/kernel/ext_manifest.h:32:10: fatal error: rimage/sof/kernel/ext_manifest.h: No such file or directory
   32 | #include <rimage/sof/kernel/ext_manifest.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

**Review by galak — COMMENTED** [review]
Suggested something similar in https://github.com/zephyrproject-rtos/zephyr/issues/31143

**Comment by nashif** [comment]
> When I try this locally I get:

yes, you need to call `west sof update`. I suggest we keep the sample and just skip it for now until west is capable of fetching sub-modules automatically when invoking `west update`.

File: samples/audio/sof/sample.yaml [hunk 1]
```diff
@@ -7,5 +7,6 @@ tests:
   sample.audio.sof:
     tags: sof
     build_only: true
+    platform_allow: intel_adsp_cavs15
```
Comment (galak) [inline samples/audio/sof/sample.yaml:10]: This should add all the other cavs platforms that are supported in `samples/audio/sof/boards`

**Comment by nashif** [comment]
obsoleted by #31171

**Pull request closed without merging**